app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -196,6 +197,25 @@ def enabled_repos
       respond_for_show :resource => result
     end
 
+    api :PUT, "/systems/:id/deb_package_profile", N_("Update installed deb packages")
```

Comment:
Newer versions of subscription-manager (which I would suggest to be the minimum for debian support) will send us a "combined profile" which potentially contains enabled repositories, rpm data, so forth.

We have an action that supports handling that particular API format already:
https://github.com/Katello/katello/blob/7474c9d72928d3d539c2e0f5cd85a06017c465c2/app/lib/actions/katello/host/upload_profiles.rb#L56

https://github.com/Katello/katello/blob/7474c9d72928d3d539c2e0f5cd85a06017c465c2/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L25

I think it would be beneficial to change the approach here to use it. In the future when subscription-manager has "first-class" support for uploading deb package profiles everything will be set here from the Katello side. I realize this will require a small change to your scripts in subscription-manager, but I can test them if you'll do the updates.

Reply:
I'll try to incorporate into that endpoint. But i also like to keep the other endpoint, maybe marking it as deprecated, since the current version of the upload script in subman uses it.

Reply:
I think the subman script could easily be updated - I don't like the idea of adding an API that would immediately be deprecated. If it helps, I will review the subman change too to help it get merged quickly.

Reply:
I don't like it either. But it not so much about the change in the script that ended up being at subman repo, but about that script being deployed in production environments.

Reply:
Ok, I can work with that as long as it is deprecated and the implementation changes to use the existing ::UploadProfiles action (and you current action may also exist, but be called by ::UploadProfiles)